client/state/data-layer/wpcom/read/site/post-email-subscriptions/index.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * External Dependencies
+ */
+import { get, includes, merge } from 'lodash';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import {
+	READER_SUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_UNSUBSCRIBE_TO_NEW_POST_EMAIL,
+	READER_UPDATE_NEW_POST_EMAIL_SUBSCRIPTION,
+} from 'state/action-types';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { subscribeToNewPostEmail, unsubscribeToNewPostEmail, updateNewPostEmailSubscription } from 'state/reader/follows/actions';
+import { errorNotice } from 'state/notices/actions';
+import { getReaderFollowForBlog } from 'state/selectors';
+
+function validateParameters( params ) {
+	// the only valid param is delivery_frequency
+	const frequency = params.deliveryFrequency;
+	const validFrequencies = [
+		'instantly',
+		'daily',
+		'weekly',
+	];
+	if ( includes( validFrequencies, frequency ) ) {
+		return {
+			delivery_frequency: frequency
+		};
+	}
+	return {};
+}
+
+function requestPostEmailSubscription( { dispatch }, action, next ) {
+	dispatch( http( {
+		method: 'POST',
+		path: `/read/site/${ action.payload.blogId }/post_email_subscriptions/new`,
+		body: validateParameters( action.payload ), // have to have an empty body to make wpcom-http happy
```

Comment:
it won't change anything so ... :) 

it's very unlikely the params will be invalid.